app/src/main/java/app/grapheneos/camera/ui/activities/MainActivity.kt

```diff
@@ -1584,4 +1579,79 @@ open class MainActivity : AppCompatActivity(),
         super.onDestroy()
         SensorOrientationChangeNotifier.clearInstance()
     }
+
+    fun locationCamConfigChanged(required: Boolean) {
```

Comment:
Extract geolocation logic to another delegate class or something else. This class is already has over 1.5k strokes... Also you may create an extension or delegate for permissions checking to avoid code boilerplate